streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -779,47 +783,64 @@ public void process(final ProcessorSupplier<? super K, ? super V> processorSuppl
 
     @Override
     public <K1> KGroupedStream<K1, V> groupBy(final KeyValueMapper<? super K, ? super V, K1> selector) {
-        return groupBy(selector, Serialized.with(null, null));
+        return groupBy(selector, Grouped.with(null, null));
     }
 
     @Override
+    @SuppressWarnings("deprecation")
```

Comment:
I think if you mark this method as `@Deprecated` as well, it will also suppress the warnings, which might be better because it preserves the deprecation notice from the interface.

Reply:
ack